src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Filters/LiquidViewFilters.cs

```diff
@@ -108,5 +114,11 @@ public static FluidValue ShapeProperties(FluidValue input, FilterArguments argum
 
             return NilValue.Instance;
         }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static T ThrowArgumentException<T>(string message)
+        {
+            throw new ArgumentException(message);
+        }
```

Comment:
So, this allows methods that use it to be inlined, right?

Reply:
Yes, when there's no exception or complicated logic (for loops for example) on call site, other normal JIT rules apply though like virtual methods are never inlined.

JIT runs against the clock so some decisions are not always easy to predict. [Inlining Analyzer](https://marketplace.visualstudio.com/items?itemName=StephanZehetner.InliningAnalyzer) is a great tool to see what happens. You can of course force inlining for methods that are called in tight loops (stack building will be costly) for example, using the `AggressiveInlining` flag.